dot/sync/chain_sync.go

```diff
@@ -731,8 +731,18 @@ taskResultLoop:
 					continue taskResultLoop
 				}
 
-				blockExactIndex := blockInResponse.Header.Number - startAtBlock
-				syncingChain[blockExactIndex] = blockInResponse
+				if blockInResponse != nil {
```

Comment:
when would this ever be `nil`?

Reply:
This information is retrieved via the network, so nothing is stopping another node from sending us data with a nil block in it (either on purpose or accident) and its a case we appear to hit sometimes when running the node.

I have modified this logic to exist in the decoding of the actual response, rather than here

Reply:
I think this is a much better approach 👍